src/unit.js

```diff
@@ -0,0 +1,56 @@
+define([
+	"./core",
+	"./common/validate/parameter-presence",
+	"./common/validate/parameter-type/number",
+	"./common/validate/parameter-type/plain-object",
+	"./unit/formatter-fn",
+	"./unit/get",
+
+	"./plural"
+], function( Globalize, validateParameterPresence, validateParameterTypeNumber,
+	validateParameterTypePlainObject, unitFormatterFn, unitGet ) {
+
+/**
+ * Globalize.formatUnit( value, unit, options )
+ *
+ * @value [Number]
+ *
+ * @unit [String]: The unit (e.g "second", "day", "year")
+ *
+ * @options [Object]
+ * - form: [String] "long", "short" (default), or "narrow".
+ *
+ * Format units such as seconds, minutes, days, weeks, etc.
+ */
+Globalize.formatUnit =
+Globalize.prototype.formatUnit = function( value, unit, options ) {
+	validateParameterTypeNumber( value, "value" );
+
+	return this.unitFormatter( unit, options )( value );
+};
+
+/**
+ * Globalize.unitFormatter( unit, options )
+ *
+ * @unit [String]: The unit (e.g "second", "day", "year")
+ *
+ * @options [Object]
+ * - form: [String] "long", "short" (default), or "narrow".
+ */
+Globalize.unitFormatter =
+Globalize.prototype.unitFormatter = function( unit, options ) {
+	var compoundUnitPattern, form, unitProperties;
+
+	validateParameterPresence( unit, "unit" );
+	validateParameterTypePlainObject( options, "options" );
+
+	form = options.form || "long";
+	compoundUnitPattern = unitGet( "per", form, this.cldr ).compoundUnitPattern;
+	unitProperties = unitGet( unit, form, this.cldr );
+
+	return unitFormatterFn( unitProperties, this.pluralGenerator(), compoundUnitPattern );
```

Comment:
I'd like to suggest two changes:

1) Simplify how you get `compoundUnitPattern` (I think reusing `unitGet` for that is an overkill and kinda confusing)

```diff
- compoundUnitPattern = unitGet( "per", form, this.cldr ).compoundUnitPattern
+ compoundUnitPattern: this.cldr.main(["units", form, "per/compoundUnitPattern"])
```

2) Collapse properties:

```diff
-	var compoundUnitPattern, form, unitProperties;
+	var form, properties;
 
 	validateParameterPresence( unit, "unit" );
 	validateParameterTypePlainObject( options, "options" );
 
 	form = options.form || "long";
-	compoundUnitPattern = unitGet( "per", form, this.cldr ).compoundUnitPattern;
-	unitProperties = unitGet( unit, form, this.cldr );
+	properties = {
+		compoundUnitPattern: this.cldr.main(["units", form, "per/compoundUnitPattern"])
+		unit: unitGet( unit, form, this.cldr )
+	}
 
-	return unitFormatterFn( unitProperties, this.pluralGenerator(), compoundUnitPattern );
+	return unitFormatterFn( properties, this.pluralGenerator() );
```